### ISSUE TITLE
Lint for needless use of `Cow`

### ISSUE THREAD

**Issue by Centri3** [rust-lang/rust-clippy#11328]
### What it does

When a `Cow`'s `Borrowed` variant is only ever used for empty strings, clippy should suggest to remove the `Cow` and use `String::new()` instead.

### Advantage

- `String::new()` allocates no memory, and is essentially the same speed: https://play.rust-lang.org/?version=nightly&mode=release&edition=2021&gist=93a8649afa1b55fc84cef2829920a932 (Less instructions doesn't mean faster, but imo, even if it's slower it's worth the ergonomics)
- It's simpler.

### Drawbacks

- It may actually end up being ever so slightly slower. In some cases this may be undesirable.

### Example

```rust
if let Some(x) = x {
    x.to_string().into()
} else if let Some(y) = y {
    Cow::Borrowed("")
} else {
    "".into()
}
```

Could be written as:

```rust
if let Some(x) = x {
    x.to_string()
} else if let Some(y) = y {
    String::new()
} else {
    String::new()
}
```
